### ISSUE TITLE
CLI --rescan argument unrecognized

### ISSUE THREAD

**Issue by peleion** [chris-belcher/electrum-personal-server#160]
Running EPS (0.1.6) under Debain Buster (fresh, in a Docker) the CLI argument --rescan returns "unrecognized arguments" when trying to set up a wallet with existing transactions.

```
root@a3509f5df03e:/usr/local/bin# electrum-personal-server --rescan /usr/local/etc/electrum-personal-server/config.cfg
usage: electrum-personal-server [-h] [-l LOG] [-a] [-f LOGFMT] [--loglevel {DEBUG,INFO,WARNING,ERROR,CRITICAL}] config_file
electrum-personal-server: error: unrecognized arguments: --rescan
root@a3509f5df03e:/usr/local/bin#
```
What does work is the electrum-personal-server-rescan script that scans from genesis without problem. 

EPS runs perfectly once rescanned.

Docker setup files [here](https://github.com/peleion/eps-docker) although I really don't think this is a Docker specific issue.

ADDED: The Docker container is built by unpacking the tarball as this saves space in the container by not having to install git (which has lots of dependencies). Were there any commits after the tarball was made? I can modify it to do a github pull if this is the case - it just increases the container file size.

**Comment by peleion** [comment]
Ideally, especially for Docker use (and BTCpay integration) there would be an option like --rescan-all that would rescan from genesis without user input. A full rescan from genesis only took about 1 hour for one 2,000 address Electrum wallet on moderately slow hardware and I think this would be the most bullet-proof (and user-proof) solution.

**Comment by chris-belcher** [comment]
I think I maybe see the problem, you're using EPS version 0.1.6 but the current version is 0.1.7, and the `--rescan` argument was only added then.

**Comment by peleion** [comment]
You are correct - the tag changed from eps-vXX to electrum-personal-server-vXX and I didn't look at the _bottom_ of the list. I'll rebuild it. 

Thanks for the bitcoin-cli rescanblockchain suggestion - I'll try it instead

**Comment by peleion** [comment]
Any reason EPS can't run _while_ bitcoind is rescanning? Obviously the transactions won't show up until scanned as there's no easy way for bitcoind to signal it's done rescanning that I know of. Would  make the logic easier under Docker for it to:

1. load addresses from EPS into bitcoind wallet on initial startup
2. auto restart container to start EPS serving after the daemon exits when initial load is complete
3. then call bitcoind directly with RPC to bitcoin-cli rescanblockchain documenting that Electrum user would need to reconnect when done for historical transactions to appear.

I will put it into a startup script for the container if this is correct. If EPS needs to restart after rescan I'll just put in an arbitrary delay time after step 3.

I've rebuilt the Docker container, just need to update the docs and push the image to DockerHub. I switched over to git pull anyway - was some weirdness with the tarball/zip file naming and will make it easier for me to maintain in the long haul.
